### PR TITLE
Implement update someday task

### LINKED ISSUE #36: implement update

### PR DESCRIPTION
**Pull request by kelvinnpc**
edit someday task feature done
parser support for edit not done
closes #36

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+/**
+ * Edits a task identified using it's last displayed index from the task manager.
+ */
+public class EditCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the task identified by the index number used in the last task listing.\n"
+            + "Parameters: INDEX (positive integer) [MORE_INDICES] ... \n"
```
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: MORE_INDICES incorrect; only one task can be edited at a time

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            + "Example: " + COMMAND_WORD + " 1 'chill for the day'";
+ 
+    public static final String MESSAGE_EDIT_TASK_SUCCESS = "Edited Task: %1$s";
+    public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in the task manager";
```
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: Note: show this msg when new name equals name of task already in manager

File: src/main/java/seedu/address/model/task/UniqueTaskList.java [hunk 3]
```diff
@@ -73,6 +76,36 @@ public boolean remove(ReadOnlyTask toRemove) throws TaskNotFoundException {
...
+    
+    /**
+     * set the equivalent task to the specified index of the list
+     * @throws TaskNotFoundException if no such task could be found in the list.
+     */ 			
+    public boolean set(int key, SomedayTask Task) throws TaskNotFoundException {
+        assert Task != null;
+        boolean isFound = false;
+		// TODO: settle 'Task' data type
+        Task stub;
```
Comment (kabir-plod) [inline src/main/java/seedu/address/model/task/UniqueTaskList.java]: What will the stub be changed to?

File: src/main/java/seedu/address/model/task/UniqueTaskList.java [hunk 4]
```diff
@@ -73,6 +76,36 @@ public boolean remove(ReadOnlyTask toRemove) throws TaskNotFoundException {
         }
         return taskFoundAndDeleted;
     }
+    
+    /**
+     * set the equivalent task to the specified index of the list
+     * @throws TaskNotFoundException if no such task could be found in the list.
+     */ 			
+    public boolean set(int key, SomedayTask Task) throws TaskNotFoundException {
```
Comment (kabir-plod) [inline src/main/java/seedu/address/model/task/UniqueTaskList.java]: More informative method name (eg. setTaskAtIndex) would make code more readable

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 2 above)
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: Where will you check for this btw?

Comment (kelvinnpc) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: in the execute there is a try which will return duplicate task

File: src/main/java/seedu/address/model/task/UniqueTaskList.java (hunk 4 above)
Comment (kelvinnpc) [inline src/main/java/seedu/address/model/task/UniqueTaskList.java]: following how the other methods in this class are named eg. add, remove
Should I change the other methods too?

**Pull request merged**